### DISCUSSION TITLE
Authentication using API keys?

### DISCUSSION THREAD

**Discussion by naveennamani** [supabase/supabase#20475] [Feature Requests]
## I want the following feature.
Whenever a user signup, a random API key will be generated which can then be used by the user to access the rest API of the application.

## Current implementation scenario
This scheme is currently implementable using a server as follows.
* Whenever a user provides an API_KEY, the server pulls the user_id from the supabase database by logging through the service key.
* Once the API_KEY is verified and user_id is obtained, the server generates the JWT with the logged in user information by signing with the JWT token.
* The user can then use this JWT token just like the JWT token provided by supabase auth when logged in using regular methods (email, oauth etc)

## Question
* Can we implement the same scheme directly using supabase auth without involving any server?
* If yes, will there be any modifications required in the RLS?

**Comment by david-r-cox** [comment]
Related: [Support API key generation #12328](https://github.com/orgs/supabase/discussions/12328)
